packages/react-storage/src/components/StorageBrowser/actions/handlers/listLocations.ts

```diff
@@ -1,21 +1,83 @@
-import { Permission } from '../../storage-internal';
+import {
+  ListLocationsOutput,
+  Permission,
+  listCallerAccessGrants,
+} from '../../storage-internal';
+import { assertAccountId } from '../../validators';
+
+import {
+  ListHandlerOptions,
+  ListHandlerInput,
+  ListHandlerOutput,
+  ListHandler,
+} from '../types';
 
-import { ListHandlerOptions, ListHandlerOutput, ListHandler } from '../types';
 import { LocationData, LocationType } from './types';
+import { parseLocations } from './utils';
+
+const DEFAULT_PAGE_SIZE = 1000;
 
 type ExcludeType = Permission | LocationType;
 
 export interface ListLocationsHandlerOptions
   extends ListHandlerOptions<ExcludeType | ExcludeType[]> {}
 
-export interface ListLocationsHandlerInput {
-  options?: ListLocationsHandlerOptions;
-}
+export interface ListLocationsHandlerInput
+  extends ListHandlerInput<ListLocationsHandlerOptions> {}
+
 export interface ListLocationsHandlerOutput
   extends ListHandlerOutput<LocationData> {}
 
 export interface ListLocationsHandler
   extends ListHandler<ListLocationsHandlerInput, ListLocationsHandlerOutput> {}
 
-export const listLocationsHandler: ListLocationsHandler =
-  null as unknown as ListLocationsHandler;
+const shouldExclude = (
+  permission: Permission,
+  type: LocationType,
+  exclude?: ExcludeType | ExcludeType[]
+) =>
+  !exclude
+    ? false
+    : typeof exclude === 'string'
+    ? exclude === permission || exclude === type
+    : exclude.includes(permission) || exclude.includes(type);
```

Comment:
Nit: Prefer negate it to improve readability. Also can it be simplified to below?
```suggestion
  exclude
    ? typeof exclude === 'string'
      ? exclude === permission || exclude === type
      : exclude.includes(permission) || exclude.includes(type)
    : false;
```

Reply:
Kept the logic already in code base from [here](https://github.com/aws-amplify/amplify-ui/blob/feat-storage-browser/main/packages/react-storage/src/components/StorageBrowser/do-not-import-from-here/actions/listLocationsAction.ts#L27-L35)